backend/src/main/java/com/elshop/backend/product/ProductController.java

```diff
@@ -1,40 +1,58 @@
 package com.elshop.backend.product;
 
 
+import com.elshop.backend.common.AppUtils;
+import com.elshop.backend.common.S3Service;
 import com.elshop.backend.product.model.Product;
 import com.elshop.backend.product.model.request.ProductRequest;
 import com.elshop.backend.product.model.response.ProductsListResponse;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
 import javax.validation.Valid;
+import java.util.List;
 
 
 @RestController
 @RequestMapping("/api/products")
 @RequiredArgsConstructor
-@CrossOrigin
 public class ProductController {
 
-    private final ProductService service;
+    private final ProductService productService;
+    private final S3Service s3Service;
+    private final AppUtils appUtils;
 
     @GetMapping
     public ResponseEntity<ProductsListResponse> getAllProducts(@RequestParam(defaultValue = "0") int page,
                                                                @RequestParam(defaultValue = "10") int size) {
-        return new ResponseEntity<>(service.getAll(page, size), HttpStatus.OK);
+        return new ResponseEntity<>(productService.getAll(page, size), HttpStatus.OK);
     }
 
 
     @GetMapping("{id}")
     public ResponseEntity<Product> getProductById(@PathVariable String id) {
-        return new ResponseEntity<>(service.getById(id), HttpStatus.OK);
+        return new ResponseEntity<>(productService.getById(id), HttpStatus.OK);
     }
 
 
-    @PostMapping
-    public ResponseEntity<Product> createNewProduct(@RequestBody @Valid ProductRequest newProductData) {
-        return new ResponseEntity<>(service.createNewProduct(newProductData), HttpStatus.OK);
+    @PostMapping(consumes = {MediaType.APPLICATION_JSON_VALUE, MediaType.MULTIPART_FORM_DATA_VALUE})
+    public ResponseEntity<Product> createNewProduct(@RequestParam(value = "images") MultipartFile[] imageFiles,
+                                                    @RequestPart("data") @Valid ProductRequest newProductData) {
+
+        String id = appUtils.generateUuid();
+        List<String> imagesUrl = s3Service.uploadMultipleFiles(imageFiles, "products", id);
+        Product product = productService.createNewProduct(newProductData, id, imagesUrl);
+
+        return new ResponseEntity<>(product, HttpStatus.OK);
```

Comment:
There is quite a lot of logic in this controller class. Consider putting some of the operations in your service